### ISSUE TITLE
 When training the Chinese benchmark dataset with svtrv2_ctc, the loss drops to 64 and cannot go lower. How to resolve this?

### ISSUE THREAD

**Issue by AnkhXXX** [Topdu/OpenOCR#31]
When training the Chinese benchmark dataset with svtrv2_ctc, the loss drops to 64 and cannot go lower. How to resolve this?

**Comment by Topdu** [comment]
Please post the details of the config file here.

**Comment by AnkhXXX** [comment]
I only modified the dataset path and the batch size. Could you help me figure out how to solve this?
Global:
  device: gpu
  epoch_num: 20
  log_smooth_window: 20
  print_batch_step: 10
  output_dir: ./output/rec/u14m_filter/svtrv2_ctc/
  eval_epoch_step: [0, 1]
  eval_batch_step: [0, 500]
  cal_metric_during_train: True
  pretrained_model:
  checkpoints:
  use_tensorboard: false
  infer_img:
  # for data or label process
  character_dict_path: &character_dict_path ../tools/utils/benchmark_new1.txt
  # ./tools/utils/EN_symbol_dict.txt
  max_text_length: &max_text_length 25
  use_space_char: &use_space_char False
  save_res_path: ./output/rec/u14m_filter/predicts_svtrv2_ctc.txt


Optimizer:
  name: AdamW
  lr: 0.00065 # for 4gpus bs256/gpu
  weight_decay: 0.05
  filter_bias_and_bn: True

LRScheduler:
  name: OneCycleLR

  warmup_epoch: 1.5 # pct_start 0.075*20 = 1.5ep
  cycle_momentum: False

Architecture:
  model_type: rec
  algorithm: SVTRv2
  Transform:
  Encoder:
    name: SVTRv2LNConvTwo33
    use_pos_embed: False
    out_channels: 256
    dims: [128, 256, 384]
    depths: [6, 6, 6]
    num_heads: [4, 8, 12]
    mixer: [['Conv','Conv','Conv','Conv','Conv','Conv'],['Conv','Conv','FGlobal','Global','Global','Global'],['Global','Global','Global','Global','Global','Global']]
    local_k: [[5, 5], [5, 5], [-1, -1]]
    sub_k: [[1, 1], [2, 1], [-1, -1]]
    last_stage: True
    feat2d: False
  Decoder:
    name: CTCDecoder

Loss:
  name: CTCLoss
  zero_infinity: True

PostProcess:
  name: CTCLabelDecode
  character_dict_path: *character_dict_path
  use_space_char: *use_space_char

Metric:
  name: RecMetric
  main_indicator: acc
  is_filter: True


Train:
  dataset:
    name: RatioDataSet
    ds_width: True
    padding: &padding False
    data_dir_list: ['/public/home/xujing2022/xqy/CDistNet_V1/Datasets/document/document_train',
#    '../Union14M-L-LMDB-Filtered/filter_train_hard',
#    '../Union14M-L-LMDB-Filtered/filter_train_medium',
#    '../Union14M-L-LMDB-Filtered/filter_train_normal',
#    '../Union14M-L-LMDB-Filtered/filter_train_easy',
    ]
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - PARSeqAug:
      - CTCLabelEncode: # Class handling label
          character_dict_path: *character_dict_path
          use_space_char: *use_space_char
          max_text_length: *max_text_length
      - KeepKeys:
          keep_keys: ['image', 'label', 'length']
  sampler:
    name: RatioSampler
    scales: [[128, 32]] # w, h
    # divide_factor: to ensure the width and height dimensions can be devided by downsampling multiple
    first_bs: &bs 64
    fix_bs: false
    divided_factor: [4, 16] # w, h
    is_training: True
  loader:
    shuffle: True
    batch_size_per_card: *bs
    drop_last: True
    max_ratio: 12
    num_workers: 4

Eval:
  dataset:
    name: RatioDataSet
    ds_width: True
    padding: True
    data_dir_list: ['/public/home/xujing2022/xqy/CDistNet_V1/Datasets/document/document_test',
#    '../evaluation/IC13_857',
#    '../evaluation/IC15_1811',
#    '../evaluation/IIIT5k',
#    '../evaluation/SVT',
#    '../evaluation/SVTP',
    ]
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - CTCLabelEncode: # Class handling label
          character_dict_path: *character_dict_path
          use_space_char: *use_space_char
          max_text_length: *max_text_length
      - KeepKeys:
          keep_keys: ['image', 'label', 'length']
  sampler:
    name: RatioSampler
    scales: [[128, 32]] # w, h
    # divide_factor: to ensure the width and height dimensions can be devided by downsampling multiple
    first_bs: 64
    fix_bs: false
    divided_factor: [4, 16] # w, h
    is_training: False
  loader:
    shuffle: False
    drop_last: False
    batch_size_per_card: 64
    max_ratio: 12
    num_workers: 4

**Comment by AnkhXXX** [comment]
Here are my training logs. the accuracy starts at 86%, which is quite strange. Could you help me figure out how to solve this? I would greatly appreciate it!
[2024/11/21 13:10:09] openrec INFO: epoch: [1/20], global_step: 10, lr: 0.000026, acc: 0.863354, norm_edit_dis: 0.863354, num_samples: 30.000000, loss: 558.433655, avg_reader_cost: 0.15628 s, avg_batch_cost: 0.82959 s, avg_samples: 37.3, ips: 44.96217 samples/s, eta: 2 days, 3:26:11
[2024/11/21 13:10:11] openrec INFO: epoch: [1/20], global_step: 20, lr: 0.000026, acc: 0.863354, norm_edit_dis: 0.863354, num_samples: 32.000000, loss: 426.379028, avg_reader_cost: 0.00379 s, avg_batch_cost: 0.20222 s, avg_samples: 40.1, ips: 198.29553 samples/s, eta: 1 day, 7:59:09
[2024/11/21 13:10:13] openrec INFO: epoch: [1/20], global_step: 30, lr: 0.000026, acc: 0.859944, norm_edit_dis: 0.859944, num_samples: 32.000000, loss: 416.284149, avg_reader_cost: 0.00366 s, avg_batch_cost: 0.20271 s, avg_samples: 30.0, ips: 147.99547 samples/s, eta: 1 day, 1:30:44
[2024/11/21 13:10:15] openrec INFO: epoch: [1/20], global_step: 40, lr: 0.000026, acc: 0.859944, norm_edit_dis: 0.859944, num_samples: 34.000000, loss: 360.402222, avg_reader_cost: 0.00385 s, avg_batch_cost: 0.19147 s, avg_samples: 42.9, ips: 224.05520 samples/s, eta: 22:06:03
[2024/11/21 13:10:16] openrec INFO: epoch: [1/20], global_step: 50, lr: 0.000026, acc: 0.886489, norm_edit_dis: 0.886489, num_samples: 42.000000, loss: 257.327301, avg_reader_cost: 0.00368 s, avg_batch_cost: 0.14880 s, avg_samples: 40.3, ips: 270.84099 samples/s, eta: 19:31:28
[2024/11/21 13:10:18] openrec INFO: epoch: [1/20], global_step: 60, lr: 0.000026, acc: 0.885312, norm_edit_dis: 0.885313, num_samples: 34.000000, loss: 279.760315, avg_reader_cost: 0.00392 s, avg_batch_cost: 0.14828 s, avg_samples: 41.1, ips: 277.18308 samples/s, eta: 17:48:06
[2024/11/21 13:10:19] openrec INFO: epoch: [1/20], global_step: 70, lr: 0.000026, acc: 0.877500, norm_edit_dis: 0.877500, num_samples: 32.000000, loss: 291.742584, avg_reader_cost: 0.00392 s, avg_batch_cost: 0.15581 s, avg_samples: 29.1, ips: 186.76188 samples/s, eta: 16:38:15
[2024/11/21 13:10:21] openrec INFO: epoch: [1/20], global_step: 80, lr: 0.000026, acc: 0.904761, norm_edit_dis: 0.904762, num_samples: 32.000000, loss: 280.375916, avg_reader_cost: 0.00405 s, avg_batch_cost: 0.14963 s, avg_samples: 36.2, ips: 241.92963 samples/s, eta: 15:43:00
[2024/11/21 13:10:22] openrec INFO: epoch: [1/20], global_step: 90, lr: 0.000026, acc: 0.896293, norm_edit_dis: 0.896293, num_samples: 39.000000, loss: 176.670959, avg_reader_cost: 0.00407 s, avg_batch_cost: 0.14567 s, avg_samples: 45.7, ips: 313.72904 samples/s, eta: 14:58:22
[2024/11/21 13:10:24] openrec INFO: epoch: [1/20], global_step: 100, lr: 0.000026, acc: 0.886489, norm_edit_dis: 0.886489, num_samples: 39.000000, loss: 161.700256, avg_reader_cost: 0.00403 s, avg_batch_cost: 0.15006 s, avg_samples: 37.2, ips: 247.90803 samples/s, eta: 14:24:18
[2024/11/21 13:10:25] openrec INFO: epoch: [1/20], global_step: 110, lr: 0.000026, acc: 0.850446, norm_edit_dis: 0.850446, num_samples: 32.000000, loss: 157.150467, avg_reader_cost: 0.00402 s, avg_batch_cost: 0.15209 s, avg_samples: 32.0, ips: 210.39853 samples/s, eta: 13:57:06
[2024/11/21 13:10:27] openrec INFO: epoch: [1/20], global_step: 120, lr: 0.000026, acc: 0.863354, norm_edit_dis: 0.863354, num_samples: 28.000000, loss: 150.752991, avg_reader_cost: 0.00395 s, avg_batch_cost: 0.15963 s, avg_samples: 29.2, ips: 182.92070 samples/s, eta: 13:36:46
[2024/11/21 13:10:28] openrec INFO: epoch: [1/20], global_step: 130, lr: 0.000026, acc: 0.883928, norm_edit_dis: 0.883929, num_samples: 26.500000, loss: 142.065948, avg_reader_cost: 0.00407 s, avg_batch_cost: 0.15357 s, avg_samples: 35.3, ips: 229.85620 samples/s, eta: 13:17:50

**Comment by Topdu** [comment]
Metric:
  name: RecMetric
  main_indicator: acc
  # is_filter: True # delete this line for chinese text

**Comment by Topdu** [comment]
How many samples are in the training set and how many gpu's are used?

**Comment by AnkhXXX** [comment]
“I am using the document dataset from BCTR. The training set consists of 400,000 samples, and I used one GPU

**Comment by AnkhXXX** [comment]
Thank you so much for your help. I followed your instructions, and it worked!
